src/main/java/org/wso2/siddhi/plugins/idea/completion/executionelements/query/QueryCompletionContributor.java

```diff
@@ -0,0 +1,156 @@
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.siddhi.plugins.idea.completion.executionelements.query;
+
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.impl.source.tree.LeafPsiElement;
+import com.intellij.psi.tree.IElementType;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.wso2.siddhi.plugins.idea.SiddhiTypes;
+import org.wso2.siddhi.plugins.idea.psi.AnonymousStreamNode;
+import org.wso2.siddhi.plugins.idea.psi.AttributeReferenceNode;
+import org.wso2.siddhi.plugins.idea.psi.ExpressionNode;
+import org.wso2.siddhi.plugins.idea.psi.GroupByNode;
+import org.wso2.siddhi.plugins.idea.psi.HavingNode;
+import org.wso2.siddhi.plugins.idea.psi.OutputEventTypeNode;
+import org.wso2.siddhi.plugins.idea.psi.OutputRateNode;
+import org.wso2.siddhi.plugins.idea.psi.QueryInputNode;
+import org.wso2.siddhi.plugins.idea.psi.QuerySectionNode;
+import org.wso2.siddhi.plugins.idea.psi.TimeValueNode;
+
+import javax.annotation.Nonnull;
+
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addBeginingOfQueryOutputKeywords;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addHavingKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addIntoKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addOutputEventTypeKeywords;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addSuggestionsAfterQueryInput;
+import static org.wso2.siddhi.plugins.idea.completion.executionelements.query.QueryInputCompletionContributor.queryInputCompletion;
+import static org.wso2.siddhi.plugins.idea.completion.executionelements.query.QueryOutputCompletionContributor.queryOutputCompletion;
+import static org.wso2.siddhi.plugins.idea.completion.executionelements.query.QuerySectionCompletionContributor.querySectionCompletion;
+import static org.wso2.siddhi.plugins.idea.completion.util.KeywordCompletionUtils.getPreviousVisibleSiblingSkippingComments;
+
+/**
+ * Provides code completions for queries.
+ */
+public class QueryCompletionContributor {
+
+    public static void queryCompletion(@Nonnull CompletionResultSet result, PsiElement element,
+                                       PsiElement prevVisibleSibling, IElementType
+                                               prevVisibleSiblingElementType, PsiElement
+                                               prevPreVisibleSibling) {
+
+        //Suggestions related to QueryInputNode
```

Comment:
Better to add a space after `//`. Fix in other places as well.

Reply:
Fixed in commit 3aa87a6